.github/workflows/gh-pages.yml

```diff
@@ -14,21 +14,26 @@ on:
       - main
       - dev
 
+permissions:
+  contents: write
+
 jobs:
-  build:
+  build-and-deploy:
+    concurrency: ci-${{ github.ref }}
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-
-    - name: Build and Deploy
-      env:
-        NODE_VERSION: 16.x
-        CI: true
-        NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
-        GH_TOKEN: ${{secrets.GH_TOKEN}}
-      run: |
-        npm install
-        npm run build
-        npm run deploy
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+
+      - name: Install and Build 🔧
+        run: |
+          npm ci
+          npm run build
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          folder: examples
+          token: ${{ secrets.GH_PAGES_DEPLOY }}
```

Comment:
To rough out the "multi deploy" idea:
- Set a target folder based on trigger type, like I do in the [shinyapps deploy workflow](https://github.com/Gilead-BioStats/gsm.app/blob/dev/.github/workflows/shinyapps-deploy.yaml).
- Use that to set the `target-folder:` variable for this `with`. Use PR-level folders for PRs, and then probably a `dev` (in addition to the root for `main`). 
- We should also have an action to delete those PR folders when the PRs are closed, like I do with the [shinyapps terminate](https://github.com/Gilead-BioStats/gsm.app/blob/dev/.github/workflows/shinyapps-terminate.yaml) workflow.

Reply:
Filed in #433.  So basically the PR folder would be a copy/paste of a "base" website folder and the action should reference that subfolder in the deployment action?

Reply:
Yeah, that's the idea, except note that you need a folder per PR. So `{url}` would be the "deployed" version, `{url}/dev` would be the dev version, and `{url}/pr/{pr number}` would be the version for a particular pr. We probably want to make sure the base website has an index and doesn't show the path to dev and pr, but I don't know how much that matters.